### PR TITLE
PXB-2164 xbcloud doesn't return the error if the backup doesn't exist…

### PR DESCRIPTION
**Pull request by rahulmalik87**
… in s3 bucket

Probelm:
xbcloud doesn't error out if backup doesn't exist

Fix:
check if backup exists else error out

### CODE REVIEW

File: storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc [hunk 1]
```diff
@@ -774,7 +774,8 @@ bool xbcloud_download(Object_store *store, const std::string &container,
                       const std::string &backup_name) {
   std::vector<std::string> object_list;
 
-  if (!store->list_objects_in_directory(container, backup_name, object_list)) {
+  if (!store->list_objects_in_directory(container, backup_name, object_list) ||
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc:777]: so with invalid backupname, list_objects_in_directory doesn't fail (ie doesn't return false?)

Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc:777]: no it won't return false as long as it has proper access on bucket.

File: storage/innobase/xtrabackup/test/t/xbcloud.sh [hunk 2]
```diff
@@ -101,12 +101,19 @@ sakila/payment.ibd
...
     die 'xbcloud did not exit with error'
 fi
 
-if [ ! grep failed 2>$topdir/pxb-1832.log ] ; then
+if ! grep -q failed $topdir/pxb-1832.log ; then
+    die 'xbcloud did not exit with error'
+fi
+
+#PXB-2164 xbcloud doesn't return the error if the backup doesn't exist in s3 bucket
+xbcloud --defaults-file=$topdir/xbcloud.cnf get somedummyjunkbackup 2>$topdir/pxb-2164.log
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/test/t/xbcloud.sh:114]: do u remove these *.log files somewhere?

**Review by satya-bodapati — APPROVED** [review]

File: storage/innobase/xtrabackup/test/t/xbcloud.sh (hunk 2 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/test/t/xbcloud.sh:114]: they are removed as part of topdir

**Pull request merged**